low-level/sensors/impl/generic/node_hw.py

```diff
@@ -636,109 +689,88 @@ def _run_ipmitool_subcommand(self, subcommand, grep_args=None, out_file=subproce
-            channel_info = self.LAN_CHANNEL_INFO
-
-        # kcs_interface_alert=True and self.lan_interface_alert=True when fault alert raised
-        # retcode == 0 means ipmitool cmd executed successfully
-        # when both condition satisfied raise falut resolved alert
-        if self.channel_err:
-            if retcode == 0:
-                self.channel_err = False
-                channel_status = "Server BMC is reachable"
-            if self.kcs_channel_err and retcode == 0:
-                alert_type = "fault_resolved"
-                resource_type = "node:bmc:interface:kcs"
-                self.kcs_interface_alert = False
-                self.kcs_channel_err = False
-        if self.lan_cmd_retcode == 0 and self.lan_fault == "fault":
-            alert_type = "fault_resolved"
-            resource_type = "node:bmc:interface:rmcp"
-            self.lan_interface_alert = False
-            raise_fault_resolved_lan = True
-            self.lan_channel_err = False
-            self.lan_fault = None
-            store.put(self.lan_fault,self.LAN_ALERT)
+    def _check_channel_error(self, err, retcode):
+        """Check BMC accessibility for active_interface."""
 
-        if (self.kcs_interface_alert or self.lan_interface_alert) and not self.channel_err:
+        if self.active_bmc_if == system:
+            self.check_kcs_channel(err, retcode)
+        elif self.active_bmc_if in BMCInterface.LAN_IF.value:
+            self.check_lan_channel(err, retcode)
+        if all([self.system_fault, self.lan_fault, self.lanplus_fault]) == "fault":
             self.channel_err = True
-            channel_status = "Server BMC is unreachable, possible cause: " + err
-            if self.lan_interface_alert and not self.lan_channel_err and self.lan_fault is None:
-                alert_type = "fault"
-                resource_type = "node:bmc:interface:rmcp"
-                self.lan_channel_err = True
-                logger.warning("BMC is unreachable through lan interface ip, \
-                                ipmitool fallback to KCS interface if local server is being monitored")
-                self.active_bmc_if = self.ipmi_client.SYSTEM_IF
-                store.put(self.active_bmc_if,self.ACTIVE_BMC_IF)
-                if self.lan_fault is None:
-                    self.lan_fault = alert_type
-                    store.put(self.lan_fault, self.LAN_ALERT)
-            elif self.kcs_interface_alert and not self.kcs_channel_err:
-                resource_type = "node:bmc:interface:kcs"
-                alert_type = "fault"
-                self.kcs_channel_err = True
-
-        if self.CHANNEL_INFO:
-            channel_info = self.CHANNEL_INFO
-
-        if self.active_bmc_if != self.ipmi_client.LAN_IF and raise_fault_resolved_lan:
-            channel_info = self.LAN_CHANNEL_INFO
+        elif any([self.system_fault, self.lan_fault, self.lanplus_fault]) != "fault":
+            self.channel_err = False
```

Comment:
channel_err should just set or unset while detecting fault for active interface. We dont need to maintain multiple variables carrying different interface fault, but just one as interface_fault and active_interface can hold ipmi interface type which is being used

Reply:
As mentioned in above comment, we switched back to system interface in case of lan fault, if bmc is not accessible from both(system/lan) interface then channel_err = True, and if it is accessible from any one of the interface, then channel_err set as false.
we need to use atleast two variable here, one for system fault and for lan/lanplus fault we can use same var.

Reply:
right, two variables we can use

Reply:
ok, add comments around relevant code

Reply:
done.